core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -479,4 +501,31 @@ object LogConfig {
     logProps.put(MessageDownConversionEnableProp, kafkaConfig.logMessageDownConversionEnable: java.lang.Boolean)
     logProps
   }
+
+  def shouldIgnoreMessageFormatVersion(interBrokerProtocolVersion: ApiVersion): Boolean =
+    interBrokerProtocolVersion >= KAFKA_3_0_IV1
```

Comment:
I wonder if we should introduce `KAFKA_3_0_IV2` to be safe here. I don't think anyone already uses `KAFKA_3_0_IV1` but what would happen if one does?

Reply:
You mean if someone is using trunk with IBP 3.0-IV1, is using the old message format and doesn't want the new behavior? Seems very unlikely, but it's a fair point. I can introduce `IV2` to be avoid it altogether.

Reply:
On second thought, this is going to be very confusing since we already have `3.1-IV0`. I think we're better off just going with `3.0-IV1`. The change here is benign in general and the IBP check is us being super careful. The number of people using IBP `3.0-IV1` is probably 0, but if there are some, they are not likely to be using the ancient record formats 0 and 1 in such clusters for new records.

Reply:
I added a comment to `ApiVersion`

Reply:
Yeah, I do agree with you. I don't think that anyone already issues `3.0-IV1` anyway.